### DISCUSSION TITLE
Proposal: Ability to hide diagnostic information in the status bar

### DISCUSSION THREAD

**Discussion by zzzint** [zed-industries/zed#14689] [Feature Requests]
## Problem Statement:
The constantly updating diagnostic messages and icons in the status bar may be distracting for some users, conflicting with [Zed's vision of an unobtrusive editor](https://zed.dev/about) 

<img width="448" alt="Screenshot 2024-07-17 at 5 21 10 PM" src="https://github.com/user-attachments/assets/e0873b99-1779-44a7-9d7a-769e89d9ed09">

<img width="600" alt="Screenshot 2024-07-17 at 12 05 21 PM" src="https://github.com/user-attachments/assets/6f8aebd3-e74c-46f1-a780-73cb5f5a7ff2">

## Proposed Solution:
Introduce two new settings under the existing `diagnostics` key:

- `show_status_bar_indicator`: Toggle visibility of diagnostic-related icons in the status bar
- `show_status_bar_message`: Toggle visibility of cursor-related diagnostic messages in the status bar

**Comment by fyodorio** [comment]
I hate "me-too"s but let's try to make it happen please 😅 I love Zed and would really love to be able to fully customize it, these annoying warnings cause me pain and anxiety (like, I _know_ my warnings, come on, don't remind me about these buggers please — at least outta context 😅)

And yeah, the idea of a disappearing editor is so nice 👍 

_Also this discussion should maybe link to [this issue](https://github.com/zed-industries/zed/issues/5120), I hope it won't be too irrelevant..._

**Reply by zzzint** [reply]
"me-too"s very welcome!

I'm more than happy to open a PR with the changes and will likely do so this weekend, just following the contributor etiquette by getting a discussion going first.

**Reply by fyodorio** [reply]
It would probably make sense to make the property name consistent (`button`) with other similar controls though

<img width="200" alt="Screenshot 2024-07-19 at 22 31 30" src="https://github.com/user-attachments/assets/8014ea15-d16a-4214-90ca-e44603fce8f9">

**Comment by xoich** [comment]
Sure, but what about the diagnostics in the buffer itself? Isn't that even worse? 

I asked this here: https://github.com/zed-industries/zed/discussions/14913

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
